--- a/llvm/include/llvm/ExecutionEngine/SectionMemoryManager.h
+++ b/llvm/include/llvm/ExecutionEngine/SectionMemoryManager.h
@@ -109,7 +109,7 @@
   ///
   /// If \p ReserveAlloc is true all memory will be pre-allocated, and any
   /// attempts to allocate beyond pre-allocated memory will fail.
-  SectionMemoryManager(MemoryMapper *MM = nullptr, bool ReserveAlloc = false);
+  SectionMemoryManager(MemoryMapper *MM = nullptr, bool ReserveAlloc = true);
   SectionMemoryManager(const SectionMemoryManager &) = delete;
   void operator=(const SectionMemoryManager &) = delete;
   ~SectionMemoryManager() override;
